### PR TITLE
Upgrade Kafka Clients/Wireprotocol model to 2.8.2

### PR DESCRIPTION
**Pull request by yzang2019**
### Motivation

Cherry pick https://github.com/datastax/starlight-for-kafka/commit/b26c615d8310dd597140b79fcb6b762aaaa699f5

### Modifications

Upgrade Kafka Clients/Wireprotocol model to 2.8.2

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Comment by BewareMyPower** [comment]
> Error:  testAlterClientQuotas(io.streamnative.pulsar.handlers.kop.KafkaAdminTest)  Time elapsed: 0.444 s  <<< FAILURE!
> java.util.concurrent.ExecutionException: org.apache.kafka.common.errors.UnsupportedVersionException: The version of API is not supported.

There is a failed test

**Comment by yzang2019** [comment]
This is probably not needed any more?

**Comment by BewareMyPower** [comment]
Not sure, but you should give a clear explanation if the change affects any existing test.

**Comment by Demogorgon314** [comment]
@yzang2019 Can you rebase to the master branch?

File: pom.xml [hunk 1]
```diff
@@ -48,7 +48,7 @@
     <!-- dependencies -->
     <jackson.version>2.14.0</jackson.version>
     <jackson-databind.version>2.13.4.2</jackson-databind.version>
-    <kafka.version>2.8.0</kafka.version>
+    <kafka.version>2.8.2</kafka.version>
```
Comment (Demogorgon314) [inline pom.xml:51]: Can you help fix the unit test?

Change like this:
```java
    @Test(timeOut = 30000, expectedExceptions = UnsupportedVersionException.class)
    public void testAlterClientQuotas() throws ExecutionException, InterruptedException {

        // TODO: Support alter client quotas by reuse pulsar topic policy.
        kafkaAdmin.alterClientQuotas(Collections.singleton(
                new ClientQuotaAlteration(
                        new ClientQuotaEntity(Maps.of(ClientQuotaEntity.CLIENT_ID, "test_client").build()),
                        List.of(new ClientQuotaAlteration.Op("producer_byte_rate", 1024.0))))).all().get();
    }

```

**Pull request closed without merging**